vms/platformvm/api/static_service.go

```diff
@@ -72,16 +73,11 @@ func (utxo UTXO) Compare(other UTXO) int {
 	return utxoAddr.Compare(otherAddr)
 }
 
-// TODO: Refactor APIStaker, APIValidators and merge them together for
-//       PermissionedValidators + PermissionlessValidators.
-
-// APIStaker is the representation of a staker sent via APIs.
+// Staker is the representation of a staker sent via APIs.
 // [TxID] is the txID of the transaction that added this staker.
-// [Amount] is the amount of tokens being staked.
-// [StartTime] is the Unix time when they start staking
 // [Endtime] is the Unix time repr. of when they are done staking
 // [NodeID] is the node ID of the staker
-// [Uptime] is the observed uptime of this staker
+// [Weight] is the validator weight (stake) when sampling validators
 type Staker struct {
 	TxID      ids.ID      `json:"txID"`
```

Comment:
TxID and EndTime will have zero values for L1 validators. I did not want to break any client implementation. If we think it's weird to show these for L1 validators I can set them to pointers. We can even break `Staker` into `BaseStaker` and `Staker` and move those fields there.

Reply:
Maybe just add a deprecated comment for TxID?

Reply:
These will still apply for non L1 stakers (subnets/primary validators)